### PR TITLE
Community Redirects

### PR DESCRIPTION
**Pull request by timolegros**
## Link to Issue
Closes: #6863

## Description of Changes
- Adds the `Communities.redirect` column which can contain an old community id
- Adds `app.config.redirects` which aggregates any defined `Communities.redirect` values
- Adds checks in create community flows on client + API to ensure the new community id does not match any of the redirect values.

## Test Plan
1. Navigate to admin-panel
2. Update any communities id
3. Access the old community id directly by URL (ensure you are properly redirected
4. Click on a thread in the new community
5. Edit URL -> change new community id to old community id and ensure redirect to thread works
6. Search for the old community in the search bar and ensure it doesn't show up
7. Search for the new community in the search bar and ensure it only shows up once
8. Enabled community directory for the same chain node as the community whose id you updated
9. Search for the old/new community ids in the directory

## Deployment Plan

1. 

## Other Considerations

-

### CODE REVIEW

**Review by mzparacha — APPROVED** [review]
LGTM

**Comment by timolegros** [comment]
> 1. warning should be updated as redirects should work now
>    image
> 2. If I change community id from A to B, and then from B to C, redirect only works from B to C, but does not from A to B. Is this expected?
> 
> besides, lgtm

1. Fixed
2. This is correct -> chains of redirects are not supported with the current implementation but we can easily expand the logic to fit this scenario. That said, I'm not sure that we need this right now. @jnaviask should we support chains of redirects?

**Comment by timolegros** [comment]
@Rotorsoft @rbennettcw @kurtisassad Even though 2 people have reviewed this if one of y'all could review the server-side changes I would appreciate it.

**Review by rbennettcw — APPROVED** [review]

File: packages/commonwealth/server/controllers/server_communities_methods/update_community_id.ts [hunk 1]
```diff
@@ -30,6 +30,25 @@ export async function __updateCommunityId(
   if (!originalCommunity) {
     throw new AppError('Community to rename not found!');
   }
+
```
Comment (Rotorsoft) [inline packages/commonwealth/server/controllers/server_communities_methods/update_community_id.ts:33]: small detail, but we do a single findOne with redirect:community_id OR id:new_community_id and format the error msg according to the result

**Review by Rotorsoft — APPROVED** [review]

**Comment by jnaviask** [comment]
Re:
> should we support chains of redirects?

I see no need to for the time being. The expectation is that this feature will be used for primarily large communities, as it requires explicit growth team intervention, and it's unlikely they'll ask for multiple changes with the expectation that redirects will be preserved.

**Pull request merged**